util/stopper_test.go

```diff
@@ -237,3 +236,52 @@ func TestStopperClosers(t *testing.T) {
 		t.Errorf("expected true & true; got %t & %t", tc1, tc2)
 	}
 }
+
+func TestStopperNumTasks(t *testing.T) {
+	s := NewStopper()
+
+	var tasks []PendingTask
+	for i := 0; i < 3; i++ {
+		tasks = append(tasks, s.StartTask())
+		if numTypes, numTasks := len(s.tasks), s.NumTasks(); numTypes != 1 || numTasks != i+1 {
+			t.Errorf("stopper should have %d running tasks, got %d / %+v", i+1, s.NumTasks(), s.tasks)
+		}
+		m := s.RunningTasks()
+		if len(m) != 1 {
+			t.Fatalf("expected exactly one task map entry: %+v", m)
```

Comment:
I think its better to prefer Fatal over Error in order to speed up test runs, unless you're certain that continuing the test may yield more useful information.  For instance, if there are several orthagonal test cases being run in a loop, or if you perform several assertions on a complicated return value, it's probably appropriate to use Errorf.

However, these assertions appear to indicate that test setup has failed in some way; there's no way the rest of the test could be relied upon, so its appropriate to fail fast.